Autogenerated relation names in dialog instead of ids

signedav opened the pull request:
The autogenerated ids are quite long and this leads that they where very unconfortable to use in the configuration of relation reference in the attribute type widget of layer properties - attribute form. So now the names are used in the combo instead of the ids. This effects only the GUI.

Done it in 

Fix: #17441

signedav committed:
Autogenerated relation names in dialog instead of ids

The autogenerated ids are quite long and this leads that they where very unconfortable to use in the configuration of relation reference in the attribute type widget of layer properties - attribute form. So now the names are used in the combo instead of the ids. This effects only the GUI.

Fix: #17441

DelazJ commented:
duplicate of #5617 ?

signedav commented:
It's for 2_18.

m-kuhn commented on src/gui/editorwidgets/qgsrelationreferenceconfigdlg.cpp:
```diff
@@ -47,7 +47,7 @@ QgsRelationReferenceConfigDlg::QgsRelationReferenceConfigDlg( QgsVectorLayer* vl
   Q_FOREACH ( const QgsRelation& relation, vl->referencingRelations( fieldIdx ) )
   {
-    mComboRelation->addItem( QString( "%1 (%2)" ).arg( relation.id(), relation.referencedLayerId() ), relation.id() );
+    mComboRelation->addItem( QString( "%1 (%2)" ).arg( relation.name(), relation.referencedLayerId() ), relation.id() );
```
Names can be left empty, I'd say we either need to enforce that a name is never empty or fallback to the id in case the name is empty.

nyalldawson replied:
How about simplifying the auto generated IDs too? Trimming the unique id to a couple of characters should be ok

signedav replied:
Thanks guys. I prefer the fallback than the "never empty" because of the backwards compatibility with older projects.
Simplifying the ids is a nice idea too. I'll check the possibilities. Do you mean in general or just here to display in the combo?

m-kuhn replied:
In general

signedav replied:
The relationid consists of the field names and the ids of the referencing and the referenced layer. These layer ids could be quite long. So another question appears - should the layer ids be simplified  in general or should we simplify them only as component of the relationid?
(the relationid is anyway unique because of a suffix function)

m-kuhn replied:
To keep this pull request in a state where it can be merged without much discussion, I would avoid changing the ids themselves.
The other question would need a wider discussion (but sounds like a good plan)

signedav committed:
Use id in gui if name is empty and crop layerid-part in the generated id

The name can be empty, so use the id instead in the combo box. To avoid so long ids, the part taken from the layerid in the relation id is cropped to 10 signs. They are unique anyway because there is used a suffix counting up.

signedav commented:
New commit. Now the relation id is used in the combo, in case the relation name is empty.
And the layer id part in the relation id is only the first 10 signs. 10 signs is not that long but you still can recognize the layers mostly. If okay, I'll fix the same in the 3.0...
